teps/0157-enhance-results-cli.md

```diff
@@ -0,0 +1,152 @@
+- As a `Tekton user`, I want to use the CLI as a plugin within `tkn` to keep all pipeline-related commands under a single tool.
+
+## Requirements
+
+### Functional Requirements
+
+- Retrieve a `list` of PipelineRuns and TaskRuns from Tekton Results.
+- Describe a `specific` PipelineRun or TaskRun with details.
+- Fetch `logs` of a PipelineRun or TaskRun.
+- Authenticate with the `Tekton Results API` using token-based authentication.
+- Support filtering results based on `namespace` (`-n` flag).
+
+### Non-Functional Requirements
+
+- Consistent command syntax, following `tkn` CLI conventions.
+- Formatted output, similar to the tkn CLI.
+- Provide clear and actionable `error messages`.
+
+## Proposal
+
+This proposal introduces a new CLI tool, `tkn-results`, designed to simplify interaction with Tekton Results. The CLI will provide commands to list, describe, and retrieve logs for `PipelineRuns` and `TaskRuns` stored in the Results API. Additionally, it will include authentication support to ensure secure access. The goal is to improve usability and efficiency over the existing CLI while maintaining a familiar command structure for Tekton users. The CLI will also be designed to work as a `plugin` to `tkn`, allowing users to execute commands seamlessly from within their existing Tekton CLI environment.
+
+## Design Details
+
+### Commands
+
+#### Configuration
+
+```sh
+tkn-results config set        # Set authentication and configuration parameters
+```
+
+#### Fetching PipelineRuns and TaskRuns
+
+```sh
+tkn-results pr ls -n <namespace>  # Get list of PipelineRuns in a namespace  
+tkn-results tr ls -n <namespace>  # Get list of TaskRuns in a namespace  
+```
+
+#### Describing Runs
+
+```sh
+tkn-results pr describe <pr-name> -n <namespace>  # Get details of a specific PipelineRun  
+tkn-results tr describe <tr-name> -n <namespace>  # Get details of a specific TaskRun  
+```
+
+#### Fetching Logs
+
+```sh
+tkn-results pr logs <pr-name> -n <namespace>  # Fetch logs of a specific PipelineRun  
+tkn-results tr logs <tr-name> -n <namespace>  # Fetch logs of a specific TaskRun  
+```
+
+### Architecture
+
+- The CLI will interact directly with `Tekton Results API` to fetch records.
+
+- Retrieved records will be parsed and structured as `PipelineRun` or `TaskRun` objects, and their details will be formatted and displayed on the CLI.
+
+- It will use `token-based authentication` to secure API requests.
+
+### Dependencies
+
+- Golang-based CLI framework (e.g., `cobra` for command parsing).
+- Tekton Results API v1 as the backend.
+
+## Alternatives Considered
+
+1. `Custom API Scripts`: Users could write their own scripts, but this lacks standardization and usability.
+
+## Impact
+
+### Backward Compatibility
+
+Existing `tkn-results` CLI commands should not break. We should add a deprecation message to the old commands to inform users about the new structure and provide guidance on transitioning to the updated CLI.
```

Comment:
We can probably hide these commands under debug or something. I found these to be helpful while debugging sometime.

Reply:
Yes, we will not remove the existing commands, as users might be relying on those commands.